www/search.py

```diff
@@ -8,14 +8,15 @@
 import json
 import random
 import cgi
+import re
 
 def getlimit(request):
 	offset = 0
 	limit = 10
 	if "offset" in request:
 		offset = int(request["offset"])
 	if "limit" in request:
-		limit = min(64, int(request["limit"]))
+		limit = min(6400, int(request["limit"]))
```

Comment:
thats bad as it heavily simplifies DoS.

Reply:
or: better explain why you want to remove the limit at all.

Reply:
A user that's doing spring-related dev work apparently needed this.

https://discord.com/channels/223585969956323328/223585969956323328/945073179007811604

Kal'tist — 02/20/2022
my lobby doesnt allow that neither
but my server needs all those map entries
any endpoint or api that lists all the map file names would also suffice 
really
Kal'tist — 02/20/2022
hmm may I ask what's blocking this?
@raaar

(i made the change)

Kal'tist — 02/20/2022
...
ty for the help!!           <---- user was happy

Reply:
I don't see how adding a different function or parameters to get the full list solves the DoS as the malicious user can spam that request instead. I think there should be something else tracking spammy sources and blocking them, if necessary.

Reply:
It makes much more sense to get back all 763 items in the result when you click the link to search maps with a given keyword, like "Large (763)"

testing a few requests:
- default/latest entries			280ms
- all games, 487 results			~800ms
- getting map keywords summary		267ms
- get map details					216ms
- get all "medium" maps, 900+ results	2.3s
- all maps, no keyword, 3k+ results		2.87s
- files named desert, 60ish results		289ms
- all maps, "latest only", 2k+ results	3.4s
- all maps, "latest only", 12x12 to 16x16, 687 results	1.25s
	
the ones that seem worse still only take about 3s, even with thousands of results compared to about 10% as much if limited to a few dozen. And the delay seems to be mostly from python processing the large result set rather than the query itself.

this can be optimized in a number of ways but imo that should go into later PR